ocean_lib/assets/test/test_asset_resolver.py

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from ocean_lib.assets.asset import Asset
+from ocean_lib.assets.asset_resolver import resolve_asset
+from tests.resources.ddo_helpers import wait_for_ddo
+from tests.resources.helper_functions import get_publisher_wallet
+
+
+def test_resolve_asset(publisher_ocean_instance, metadata):
+    publisher = get_publisher_wallet()
+    metadata_copy = metadata.copy()
+
+    asset = publisher_ocean_instance.assets.create(metadata_copy, publisher)
+    wait_for_ddo(publisher_ocean_instance, asset.did)
+    assert asset is not None, "The asset is not cached."
+
+    resolved_asset = resolve_asset(
+        asset.did,
+        publisher_ocean_instance.config.metadata_cache_uri,
+        asset.data_token_address,
+    )
+    assert isinstance(asset, Asset), "The asset does not have Asset instance."
+    assert isinstance(resolved_asset, Asset), "The asset does not have Asset instance."
+    assert (
+        resolved_asset.did == asset.did
+    ), "Resolve asset function call is unsuccessful."
+
+
+def test_bad_resolved_asset(publisher_ocean_instance, metadata):
+    publisher = get_publisher_wallet()
+    metadata_copy = metadata.copy()
+
+    asset = publisher_ocean_instance.assets.create(metadata_copy, publisher)
+    wait_for_ddo(publisher_ocean_instance, asset.did)
+    assert asset is not None, "The asset is not cached."
+
+    try:
```

Comment:
I would suggest using assert pytest.raises() instead of a try catch. Although I do want to keep the message checking somehow, because that's precise and good.